tests/Doctrine/Tests/ORM/Functional/GH10752Test.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @Group GH10752
+ */
+class GH10752Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10752Order::class,
+            GH10752Promotion::class,
+        ]);
+    }
+
+    public function testThrowExceptionWhenRemovingPromotionThatIsInUse(): void
+    {
+        if (! $this->_em->getConnection()->getDatabasePlatform()->supportsForeignKeyConstraints()) {
+            self::markTestSkipped('Platform does not support foreign keys.');
+        }
+
+        $order     = $this->createOrder();
+        $promotion = $this->createPromotion();
+
+        $order->addPromotion($promotion);
+
+        $this->_em->persist($order);
+        $this->_em->persist($promotion);
+        $this->_em->flush();
+
+        $this->_em->remove($promotion);
+
+        $this->expectException(ForeignKeyConstraintViolationException::class);
+        $this->_em->flush();
+    }
+
+    public function testThrowExceptionWhenRemovingPromotionThatIsInUseAndOrderIsNotInMemory(): void
+    {
+        if (! $this->_em->getConnection()->getDatabasePlatform()->supportsForeignKeyConstraints()) {
+            self::markTestSkipped('Platform does not support foreign keys.');
+        }
+
+        $order     = $this->createOrder();
+        $promotion = $this->createPromotion();
+
+        $order->addPromotion($promotion);
+
+        $this->_em->persist($order);
+        $this->_em->persist($promotion);
+        $this->_em->flush();
+
+        $this->_em->clear();
+
+        $promotion = $this->_em->find(GH10752Promotion::class, $promotion->getId());
+        $this->_em->remove($promotion);
+
+        $this->expectException(ForeignKeyConstraintViolationException::class);
+        $this->_em->flush();
+    }
+
+    private function createOrder(): GH10752Order
+    {
+        return new GH10752Order();
+    }
+
+    private function createPromotion(): GH10752Promotion
+    {
+        return new GH10752Promotion();
+    }
```

Comment:
What is the point of these methods?

Reply:
*shrugs* I just picked that from #10752 without thinking about it

Reply:
I'll change it